### ISSUE TITLE
Switch license to Rust license?

### ISSUE THREAD

**Issue by Manishearth** [rust-lang-nursery/rust-clippy#2885]
Especially given that uplifts are happening, we should consider switching over to Rust's MIT/Apache v2 license. If we want we can keep MPL to have it triple licensed.

The process involved pinging everyone who has committed to the project and asking them for consent.

Thoughts?

cc @oli-obk @llogiq @mcarton @flip1995 @phansch @birkenfeld

**Comment by oli-obk** [comment]
Feels like the safest variant. I hope we can reach everyone. This has been done before, are there any scripts around that could be used?

**Comment by mcarton** [comment]
We have 204 contributors now, so it might get a little tricky. We don't need all of them to agree though, very small contributions (eg. typo fixing) aren't protected like that. But it's hard to choose between small contribution or not. Also, IANAL.
What we could at least do is switch the license for *future commits*. All contributions until now would stay on the single licence we have now, and all future contributions would be doubly licenced. This would at least cover most new lints.

Anyway, as I just happened to come here a few minutes after this was created, and I'm not as involved as before (and don't even have time to read the dozens of emails I get from GitHub everyday), I'd better speak now:

> I license past and future contributions under the dual MIT/Apache-2.0 license, allowing licensees to chose either at their option.

**Comment by mcarton** [comment]
Some time ago, there has been a wave of issues opened all over Rust projects' GitHub about licence changes with a generated list of all contributors. I've seen the template evolve over time. Any idea who did that?
(Also, why was Clippy spared?)

**Comment by CAD97** [comment]
[When sorted by additions, the GitHub contributor list gets to #100 with 2++](https://github.com/rust-lang-nursery/rust-clippy/graphs/contributors?from=2014-11-16&to=2018-07-16&type=a).

IANAL, but I'm thinking nobody will complain much if you manage to get all 100 of those contributors to OK a re-license.

**Comment by cmr** [comment]
https://github.com/cmr/relicense-assistant is everything I used to make that wave. Clippy was spared because I only targeted MIT xor Apache-2.0 projects. I wasn't interested in getting people to substantially change their license.

**Comment by cmr** [comment]
Note that for changing the license, you'll want to change the text to something along the lines of:

`I license past and future contributions under the dual MIT/Apache-2.0 license, and MPL-2.0, allowing licensees to chose any at their option.`

This keeps the MPL-2.0 status quo until you can actually switch.

**Comment by cavedweller** [comment]
I and some other interested parties were talking about this earlier today actually. I think it would make the most sense to make this MIT/Apache-2.0 if it's in the "core" rust ecosystem now. Should the script open an issue or add to this Issue?

**Comment by Manishearth** [comment]
Opened the relicensing issues, it's being coordinated at https://github.com/rust-lang-nursery/rust-clippy/issues/3093 

Going with simple dual licensing.

**Comment by est31** [comment]
@Manishearth I suggest that you also add a note to the README to make any *future* contributions licensed under MIT/Apache 2.0. Otherwise, you'll be always playing catch-up with the new contributors that made a PR since you opened those issues.

**Comment by est31** [comment]
To prove my point, these folks have been newly contributing under the old MPL license after the issues were opened:

* https://github.com/rust-lang-nursery/rust-clippy/commits?author=daubaris
* https://github.com/rust-lang-nursery/rust-clippy/commits?author=JoshMcguigan
* https://github.com/rust-lang-nursery/rust-clippy/commits?author=PSeitz
* https://github.com/rust-lang-nursery/rust-clippy/commits?author=yangby-cryptape

Not making any claims that this list is exhaustive.

**Comment by Manishearth** [comment]
@est31 I don't think we can, because MPL isn't as compatible with the Rust license. Which means that folks would be contributing triple licensed code to an MPL codebase and I don't really want to deal with figuring out the legality of that. I guess a readme note works but it's not 100% clear.

I'd rather have them go through the same process, or explicitly note that they allow the work to be relciensed. I plan on asking on their PRs soon, once we're close to done with the historical bunch
